### PR TITLE
RavenDB-9718:

### PR DESCRIPTION
**Pull request by maximburyak**
* Upgraded to new Jint
* Improved Jint related array processing performance
* Fixed LazyNumberValue processing inconsistency in translation to JSValue

### CODE REVIEW

File: src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs [hunk 1]
```diff
@@ -142,36 +143,76 @@ private JsValue GetPropertyValue(string key, int propertyIndex)
...
+                        if (itemType == BlittableJsonToken.Integer || itemType == BlittableJsonToken.LazyNumber)
+                        {
+                            item = TranslateToJs(null, null, blit.Item2, blit.Item1);
+                        }
+                        else
+                        {
+                            item = TranslateToJs(parent, null, blit.Item2, blit.Item1);
+                        }
+                        items[i] = new PropertyDescriptor(item, true, true, true);
+                    }
```
Comment (lahma) [inline src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs]: Boolean translation to flags has slight performance penalty, use flags directly when possible

File: src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs [hunk 2]
```diff
@@ -142,36 +143,76 @@ private JsValue GetPropertyValue(string key, int propertyIndex)
...
-                        return (bool)value;
+                        return (bool)value?JsBoolean.True:JsBoolean.False;
                     case BlittableJsonToken.Integer:
-
                         // TODO: in the future, add [numeric type]TryFormat, when parsing numbers to strings
-                        owner.RecordNumericFieldType(key, BlittableJsonToken.Integer);                                                
+                        owner?.RecordNumericFieldType(key, BlittableJsonToken.Integer);                                                
 
-                        return (long)value;
+                        return new JsNumber((long)value);
```
Comment (lahma) [inline src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs]: Casting to JsValue goes through conversion operator that has cache, allocation can be avoided for smaller numbers

Comment (lahma) [reply src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs]: https://github.com/sebastienros/jint/blob/5d408955dc904300e543d27bb12ad10c19327b84/Jint/Native/JsValue.cs#L449

File: src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs [hunk 3]
```diff
@@ -180,26 +221,35 @@ private unsafe JsValue TranslateToJs(BlittableObjectInstance owner, string key,
...
 
-            
+            public static JsValue GetJSNumberForLazyNumber(Engine engine, LazyNumberValue value)
+            {
+                // First, try and see if the number is withing double boundaries.
+                // We use double's tryParse and it actually may round the number, 
+                // But that are Jint's limitations
+                if (value.TryParseDouble(out double doubleVal))
+                {
+                    return new JsNumber(doubleVal);
```
Comment (lahma) [inline src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs]: Cast-cache could benefit here too

File: src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs (hunk 1 above)
Comment (maximburyak) [reply src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs]: This overload is "protected internal", are you sure you want to expose it?

File: src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs (hunk 2 above)
Comment (maximburyak) [reply src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs]: ok, nice

File: src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs (hunk 1 above)
Comment (lahma) [reply src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs]: I wouldn't expose to "normal" consumer, but RavenDB is quite different with separate fork :) Yes maybe can leave as-is it's really micro-optimization.

File: src/Raven.Server/Documents/Patch/ScriptRunner.cs [hunk 4]
```diff
@@ -849,15 +883,14 @@ BlittableJsonReaderObject Clone(BlittableJsonReaderObject origin)
                 if (o == null)
                     return Undefined.Instance;
                 if (o is long lng)
-                    return lng;
+                    return new JsNumber(lng);
```
Comment (lahma) [inline src/Raven.Server/Documents/Patch/ScriptRunner.cs]: caching could work for this if-branching's items too

**Pull request merged**